Model/src/main/java/org/apidb/apicommon/model/report/sequence/SequenceReporter.java

```diff
@@ -0,0 +1,152 @@
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+import org.apidb.apicommon.model.TranscriptUtil;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.glassfish.jersey.media.multipart.MultiPart;
+import org.gusdb.fgputil.IoUtil;
+import org.gusdb.fgputil.client.ClientUtil;
+import org.gusdb.fgputil.client.ResponseFuture;
+import org.gusdb.wdk.model.WdkModelException;
+import org.gusdb.wdk.model.answer.request.AnswerFormatting;
+import org.gusdb.wdk.model.answer.request.AnswerRequest;
+import org.gusdb.wdk.model.answer.request.TemporaryResultFactory;
+import org.gusdb.wdk.model.report.AbstractReporter;
+import org.apidb.apicommon.model.report.bed.BedGeneReporter;
+import org.gusdb.wdk.model.report.ReporterConfigException;
+import org.json.JSONObject;
+
+public class SequenceReporter extends AbstractReporter {
+
+  private static Logger LOG = Logger.getLogger(SequenceReporter.class);
+
+  private static final String FASTA_MEDIA_TYPE = "text/x-fasta";
+  private static final String BED_REPORTER_NAME = "bed";
+
+  private enum SequenceType {
+    genomic,
+    protein,
+    est,
+    popset;
+  }
+
+  private boolean _showInBrowser = false;
+  // data required to make sequence retrieval service request
+  private String _seqRetSvcRequestUrl;
+  private String _bedFileUrl;
+
+  @Override
+  public SequenceReporter configure(JSONObject config) throws ReporterConfigException {
+    _showInBrowser = "plain".equals(config.optString("attachmentType", "plain"));
+
+    // extract any sequence retrieval service config from this config
+    int basesPerLine;
+    if("fixed_width".equals(config.getString("sequenceFormat"))){
+      basesPerLine = config.getInt("basesPerLine");
+    } else {
+      basesPerLine = 0;
+    }
+
+    // build an answer request to save off and create a temporary result URL for
+    AnswerRequest bedReporterRequest = new AnswerRequest(_baseAnswer.getRunnableAnswerSpec(), new AnswerFormatting(BED_REPORTER_NAME, config), false);
+    String bedResultId = TemporaryResultFactory.insertTemporaryResult(_baseAnswer.getUser().getUserId(), bedReporterRequest);
+
+    // Determine needed URLs from model props
+    //   Example prop values:
+    //     LOCALHOST=https://rdoherty.plasmodb.org
+    //     SERVICE_BASE_URL=/plasmo.rdoherty/service
+    //     SEQUENCE_RETRIEVAL_SERVICE_URL=/sequence-retrieval
+    Map<String,String> modelProps = _baseAnswer.getWdkModel().getProperties();
+    String localhost = modelProps.get("LOCALHOST");
+    _bedFileUrl = localhost + modelProps.get("SERVICE_BASE_URL") + "/temporary-results/" + bedResultId;
+
+    SequenceType sequenceType;
+    try {
+      sequenceType = getSequenceTypeByRecordClassFullName(config, getQuestion().getRecordClass().getFullName());
+    } catch (WdkModelException e){
+      throw new ReporterConfigException("Could not configure reporter", e);
+    }
+
+    // FIXME: starting here with synchronous API for proof of concept; convert to async
```

Comment:
Are we planning to switch to the async for the first release?  If this gets the job done, it is much simpler than getting a job ID back and polling until some timeout....  But maybe I could add a method to ResponseFuture that takes a timeout duration.  Might be better than polling long term as well.

Reply:
I was never convinced the async stuff will be better, but chose this (maybe immature) way of not arguing about it - I just added it in to the service, but didn't actually try to use it here. I think it's gonna be good like that - potentially better - but I guess making it a future development will be fine.